metagraph/src/graph/representation/succinct/boss.cpp

```diff
@@ -639,6 +639,27 @@ edge_index BOSS::fwd(edge_index i, TAlphabet c) const {
     return select_last(target_node);
 }
 
+/**
+ * This function gets a position i reflecting the r-th occurrence of the corresponding
+ * character c in W and returns the position of the r-th occurrence of c in last.
+ */
+edge_index BOSS::fwd(edge_index i) const {
+    CHECK_INDEX(i);
+
+    auto [rank, c] = W_->inverse_select(i);
+    assert(i == 1 || c != kSentinelCode);
+    if (c == kSentinelCode) {
+        --rank;
+    } else if (c >= alph_size) {
+        c %= alph_size;
+        rank = rank_W(i, c);
+    }
```

Comment:
`inverse_select` + `rank` (which happens now in the second branch) is actually more expensive than `access` + `rank` (performed previously). Did you benchmark this for all types?

In fact, for some types (e.g., `wavelet_tree_dyn`) this PR implements `inverse_select` via the same `access` + `rank`, so the `fwd()` calls became strictly slower now ([`access` + `rank` or `access` + 2*`rank`] vs. [`access` + `rank`]).

Reply:
I've benchmarked this using alignment to a RowDiff-annotated microbial graph and this PR speeds it up about 1.2x.

The motivation is that the second branch occurs less frequently, and hence, its cost gets amortized.

Reply:
But yes, a more systematic benchmark is needed

Reply:
Using the benchmark code on the same graph:
```
-------------------------------------------------------------------
Benchmark                         Time             CPU   Iterations
-------------------------------------------------------------------
BM_BOSS_get_W_and_fwd          2.90 us         2.89 us       241361
BM_BOSS_no_get_W_and_fwd       2.48 us         2.47 us       281466
```

Reply:
On the full microbe graph, it seems to be only 4% faster
small
```
-------------------------------------------------------------------
Benchmark                         Time             CPU   Iterations
-------------------------------------------------------------------
BM_BOSS_get_W_and_fwd          6.31 us         6.31 us       105374
BM_BOSS_no_get_W_and_fwd       6.06 us         6.06 us       117376
```
stat
```
-------------------------------------------------------------------
Benchmark                         Time             CPU   Iterations
-------------------------------------------------------------------
BM_BOSS_get_W_and_fwd         0.773 us        0.773 us       898763
BM_BOSS_no_get_W_and_fwd      0.744 us        0.744 us       950109
```

Reply:
Using the new implementation of `inverse_select` in DYNAMIC on the microbe 750 graph
```
-------------------------------------------------------------------
Benchmark                         Time             CPU   Iterations
-------------------------------------------------------------------
BM_BOSS_get_W_and_fwd          4.14 us         4.13 us       168212
BM_BOSS_no_get_W_and_fwd       3.62 us         3.61 us       198161
```